commit 84f650e6323cfdb44aaa94adbab7480435427cf6
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Mon Apr 17 11:41:24 2023 +0300

    wifi: iwlwifi: mvm: remove per-STA MFP setting
    
    When we first add the STA before sending an authentication
    frame this is false. However, in some cases such as FT or
    certainly in MLO link switch scenarios, it will be true.
    
    If it happens to be true, it causes a firmware assert (in
    LMAC, 0x2528), because the same bit is used by the firmware
    for tracking this setting as well as for tracking if the
    management key has been installed, and then we get this
    assert from the firmware when installing the MFP key as it
    thinks it has already been installed.
    
    Remove the setting for now, until the firmware disentangles
    the two. We should be able to set it, in fact we should be
    setting it speculatively before authentication/association,
    to avoid processing management frames that should have been
    protected/encrypted before the key is set.
    
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>
    Signed-off-by: Gregory Greenman <gregory.greenman@intel.com>
    Link: https://lore.kernel.org/r/20230417113648.adbeb0e0bfed.I400d0ee3721dc4f294313be992d277ba4d9c88d9@changeid
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/mld-sta.c b/drivers/net/wireless/intel/iwlwifi/mvm/mld-sta.c
index 65436736f87f..fe362f621777 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/mld-sta.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/mld-sta.c
@@ -427,7 +427,6 @@ static int iwl_mvm_mld_cfg_sta(struct iwl_mvm *mvm, struct ieee80211_sta *sta,
 	struct iwl_mvm_sta_cfg_cmd cmd = {
 		.sta_id = cpu_to_le32(mvm_link_sta->sta_id),
 		.station_type = cpu_to_le32(mvm_sta->sta_type),
-		.mfp = cpu_to_le32(sta->mfp),
 	};
 	u32 agg_size = 0, mpdu_dens = 0;